controllers/operator.go

```diff
@@ -373,56 +430,62 @@ func (o *operator) estimateDeploy(deploy *appsv1.Deployment) (*appsv1.Deployment
 }
 
 func (o *operator) updateDeploy(deploy *appsv1.Deployment) (*appsv1.Deployment, error) {
-	partition, err := helpers.ParsePartitionAnnotation(deploy.Annotations[PartitionAnnotation])
+	consumerId, err := helpers.ParseIntAnnotation(deploy.Annotations[ConsumerAnnotation])
 	if err != nil {
 		return nil, err
 	}
+	partitions := o.assignments[consumerId]
 	deploy.Annotations[GenerationAnnotation] = o.observedGeneration()
 	deploy.Spec = o.consumer.Spec.DeploymentTemplate
 	for i := range deploy.Spec.Template.Spec.Containers {
 		var resources *corev1.ResourceRequirements
 		var underProvision int64
 		container := &deploy.Spec.Template.Spec.Containers[i]
 		if container.Resources.Requests.Cpu().IsZero() {
-			resources, underProvision = o.allocateResources(container, partition)
+			resources, underProvision = o.allocateResources(container, partitions)
 		} else {
-			resources, underProvision = o.updateResources(container, partition)
+			resources, underProvision = o.updateResources(container, partitions)
 		}
 		o.updateScaleAnnotations(deploy, underProvision)
 		container.Resources = *resources
-		container.Env = helpers.PopulateEnv(container.Env, &container.Resources, o.consumer.Spec.PartitionEnvKey, int(partition))
+		container.Env = helpers.PopulateEnv(container.Env, &container.Resources, o.consumer.Spec.PartitionEnvKey, partitions)
 	}
 	return deploy, nil
 }
 
-func (o *operator) constructDeploy(partition int32) *appsv1.Deployment {
-	deployLabels := make(map[string]string)
+func (o *operator) constructDeploy(consumerId int32) *appsv1.Deployment {
+	partitionIds := o.assignments[consumerId]
```

Comment:
no guarantees that o.assignments will contain consumerId. Better to extend function:
```
func (o *operator) newDeploy(consumerId int32) (*appsv1.Deployment, error) {
         partitions, ok := o. assignments[consumerId]
         if !ok {
           return fmt.Errorf("BUG: no partitions")
        }
	deploy := o.constructDeploy(consumerId, partitions)
	return o.updateDeploy(deploy)
}

func (o *operator) constructDeploy(consumerId int32, partitions []int32)
```

Reply:
This map is constructed during init and include all the partitions. `constructDeploy` is called for each missing Id by iterating on the `o.assignments` map.
I think it will always have required consumerId, at least with the current implementation. But, I guess it's safer to add the check anyways.